godot-codegen/src/models/domain/enums.rs

```diff
@@ -0,0 +1,164 @@
+    ///
+    /// Returns `None` if `self` is a bitfield.
+    pub fn unique_ords(&self) -> Option<Vec<i32>> {
+        let mut unique_ords = self
+            .enumerators
+            .iter()
+            .map(|enumerator| enumerator.value.as_enum())
+            .collect::<Option<Vec<i32>>>()?;
+
+        unique_ords.sort();
+        unique_ords.dedup();
+
+        Some(unique_ords)
+    }
+
+    /// Returns tokens representing the engine trait this enum should implement.
+    pub fn engine_trait(&self) -> TokenStream {
+        if self.is_bitfield {
+            quote! { crate::obj::EngineBitfield }
+        } else {
+            quote! { crate::obj::EngineEnum }
+        }
+    }
+
+    /// Returns the maximum index of an indexable enum.
+    ///
+    /// Return `None` if `self` isn't an indexable enum. Meaning it is either a bitfield, or it is an enum that can't be used as an index.
+    pub fn count_index_enum(&self) -> Option<usize> {
+        if self.is_bitfield {
+            return None;
+        }
+
+        // Sort by ordinal value. Allocates for every enum in the JSON, but should be OK (most enums are indexable).
+        let enumerators = {
+            let mut enumerators = self.enumerators.clone();
+            enumerators.sort_by_key(|v| v.value.to_i64());
+            enumerators
+        };
+
+        // Highest ordinal must be the "MAX" one.
+        // The presence of "MAX" indicates that Godot devs intended the enum to be used as an index.
+        // The condition is not strictly necessary and could theoretically be relaxed; there would need to be concrete use cases though.
+        let last = enumerators.last()?; // If there isn't a last we can assume it shouldn't be used as an index.
+
+        if !last.godot_name.ends_with("_MAX") {
+            return None;
+        }
+
+        // The rest of the enumerators must be contiguous and without gaps (duplicates are OK).
+        let mut last_value = 0;
+
+        for enumerator in enumerators.iter() {
+            let current_value = enumerator.value.to_i64();
+
+            if !(current_value == last_value || current_value == last_value + 1) {
+                return None;
+            }
+
+            last_value = current_value;
+        }
+
+        Some(last_value as usize)
+    }
+}
+
+#[derive(Clone)]
+pub struct Enumerator {
+    pub name: Ident,
+
+    pub godot_name: String,
+
+    // i64 is common denominator for enum, bitfield and constant values.
+    // Note that values > i64::MAX will be implicitly wrapped, see https://github.com/not-fl3/nanoserde/issues/89.
+    pub value: EnumeratorValue,
+}
+
+#[derive(Clone)]
+pub enum EnumeratorValue {
+    Enum(i32),
+    Bitfield(u64),
+}
+
+impl EnumeratorValue {
+    /// Tries to convert `self` into an enum value.
+    fn as_enum(&self) -> Option<i32> {
+        match self {
+            EnumeratorValue::Enum(i) => Some(*i),
+            EnumeratorValue::Bitfield(_) => None,
+        }
+    }
```

Comment:
The naming is a bit confusing since `Enum` is a type here and conceptually different from the ordinal number.

You probably meant "as enum, not bitfield" which could be named `as_enum_ord`? Maybe mention the "not bitfield" part in the docs.